### ISSUE TITLE
Issue when rebuilding VS Dev Container

### ISSUE THREAD

**Issue by pope-llamas0g** [Azure-Samples/azure-search-openai-demo#1749]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [X] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
> I haven't updated the repo since 2 months. I was trying to do a full pull (rebase) and then I got a pop-up message to "rebuild the container" after clicking it and waiting for a while I get the following message:

![Screenshot 2024-06-21 at 11 35 18 AM](https://github.com/Azure-Samples/azure-search-openai-demo/assets/146399031/f724e3ad-f950-4dc0-bfc9-c89a0e47f51d)

I see similar issues on dev containers still unresolved. https://github.com/microsoft/vscode-remote-release/issues/6509

### Any log messages given by the failure
> message here:
[13200 ms] * Fetched feature: powershell_3_oci version 1.1.0
[13220 ms] Start: Run: docker buildx build --load --build-context dev_containers_feature_content_source=/tmp/devcontainercli-root/container-features/0.62.0-1718955123548 --build-arg _DEV_CONTAINERS_BASE_IMAGE=mcr.microsoft.com/devcontainers/python:3.11 --build-arg _DEV_CONTAINERS_IMAGE_USER=root --build-arg _DEV_CONTAINERS_FEATURE_CONTENT_SOURCE=dev_container_feature_content_temp --target dev_containers_target_stage -f /tmp/devcontainercli-root/container-features/0.62.0-1718955123548/Dockerfile.extended -t vsc-azure-search-openai-demo-2868f436e41d9d185e18eea712c952c50496a000d57ac9f2d1ff61db04503393-features /tmp/devcontainercli-root/empty-folder
[+] Building 0.0s (1/1)                                                         
 => [internal] load .dockerignore                                          0.0s
[+] Building 10.4s (17/17) FINISHED                                             
 => [internal] load .dockerignore                                          0.0s
 => => transferring context: 2B                                            0.0s
 => [internal] load build definition from Dockerfile.extended              0.0s
 => => transferring dockerfile: 4.56kB                                     0.0s
 => resolve image config for docker.io/docker/dockerfile:1.4               1.0s
 => CACHED docker-image://docker.io/docker/dockerfile:1.4@sha256:9ba7531b  0.0s
 => [context dev_containers_feature_content_source] load .dockerignore     0.0s
 => => transferring dev_containers_feature_content_source: 2B              0.0s
 => [internal] load metadata for mcr.microsoft.com/devcontainers/python:3  0.2s
 => [dev_containers_feature_content_normalize 1/3] FROM mcr.microsoft.com  0.0s
 => [context dev_containers_feature_content_source] load from client       0.0s
 => => transferring dev_containers_feature_content_source: 68.97kB         0.0s
 => CACHED [dev_containers_target_stage 2/8] RUN mkdir -p /tmp/dev-contai  0.0s
 => CACHED [dev_containers_feature_content_normalize 2/3] COPY --from=dev  0.0s
 => CACHED [dev_containers_feature_content_normalize 3/3] RUN chmod -R 07  0.0s
 => CACHED [dev_containers_target_stage 3/8] COPY --from=dev_containers_f  0.0s
 => CACHED [dev_containers_target_stage 4/8] RUN echo "_CONTAINER_USER_HO  0.0s
 => CACHED [dev_containers_target_stage 5/8] RUN --mount=type=bind,from=d  0.0s
 => CACHED [dev_containers_target_stage 6/8] RUN --mount=type=bind,from=d  0.0s
 => CACHED [dev_containers_target_stage 7/8] RUN --mount=type=bind,from=d  0.0s
 => ERROR [dev_containers_target_stage 8/8] RUN --mount=type=bind,from=de  8.9s
------                                                                          
 > [dev_containers_target_stage 8/8] RUN --mount=type=bind,from=dev_containers_feature_content_source,source=powershell_3,target=/tmp/build-features-src/powershell_3     cp -ar /tmp/build-features-src/powershell_3 /tmp/dev-container-features  && chmod -R 0755 /tmp/dev-container-features/powershell_3  && cd /tmp/dev-container-features/powershell_3  && chmod +x ./devcontainer-features-install.sh  && ./devcontainer-features-install.sh  && rm -rf /tmp/dev-container-features/powershell_3:
#0 0.086 ===========================================================================
#0 0.086 Feature       : PowerShell
#0 0.086 Description   : Installs PowerShell along with needed dependencies. Useful for base Dockerfiles that often are missing required install dependencies like gpg.
#0 0.086 Id            : ghcr.io/devcontainers/features/powershell
#0 0.086 Version       : 1.1.0
#0 0.086 Documentation : https://github.com/devcontainers/features/tree/main/src/powershell
#0 0.086 Options       :
#0 0.086     VERSION="latest"
#0 0.086     MODULES=""
#0 0.086 ===========================================================================
#0 0.091 Attempting install from GitHub release...
#0 0.096 find: ‘/var/lib/apt/lists/*’: No such file or directory
#0 0.096 Running apt-get update...
#0 0.496 Get:1 http://deb.debian.org/debian bookworm InRelease [151 kB]
#0 0.674 Get:2 https://dl.yarnpkg.com/debian stable InRelease [17.1 kB]
#0 0.874 Get:3 http://deb.debian.org/debian bookworm-updates InRelease [55.4 kB]
#0 0.887 Get:4 https://dl.yarnpkg.com/debian stable/main arm64 Packages [10.9 kB]
#0 1.008 Get:5 http://deb.debian.org/debian-security bookworm-security InRelease [48.0 kB]
#0 1.029 Get:6 https://dl.yarnpkg.com/debian stable/main all Packages [10.9 kB]
#0 1.140 Get:7 http://deb.debian.org/debian bookworm/main arm64 Packages [8685 kB]
#0 1.906 Get:8 http://deb.debian.org/debian bookworm-updates/main arm64 Packages [13.7 kB]
#0 2.023 Get:9 http://deb.debian.org/debian-security bookworm-security/main arm64 Packages [157 kB]
#0 2.535 Fetched 9149 kB in 2s (3773 kB/s)
#0 2.535 Reading package lists...
#0 2.789 Reading package lists...
#0 3.038 Building dependency tree...
#0 3.098 Reading state information...
#0 3.200 curl is already the newest version (7.88.1-10+deb12u5).
#0 3.200 ca-certificates is already the newest version (20230311).
#0 3.200 gnupg2 is already the newest version (2.2.40-1.1).
#0 3.200 dirmngr is already the newest version (2.2.40-1.1).
#0 3.200 libc6 is already the newest version (2.36-9+deb12u7).
#0 3.200 libgcc-s1 is already the newest version (12.2.0-14).
#0 3.200 libgssapi-krb5-2 is already the newest version (1.20.1-2+deb12u1).
#0 3.200 libstdc++6 is already the newest version (12.2.0-14).
#0 3.200 libuuid1 is already the newest version (2.38.1-5+deb12u1).
#0 3.200 libuuid1 set to manually installed.
#0 3.200 zlib1g is already the newest version (1:1.2.13.dfsg-1).
#0 3.200 libicu72 is already the newest version (72.1-3).
#0 3.200 The following NEW packages will be installed:
#0 3.200   libunwind8
#0 3.449 0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
#0 3.449 Need to get 47.4 kB of archives.
#0 3.449 After this operation, 360 kB of additional disk space will be used.
#0 3.449 Get:1 http://deb.debian.org/debian bookworm/main arm64 libunwind8 arm64 1.6.2-3 [47.4 kB]
#0 4.035 Fetched 47.4 kB in 0s (126 kB/s)
#0 4.046 Selecting previously unselected package libunwind8:arm64.
(Reading database ... 30264 files and directories currently installed.)
#0 4.055 Preparing to unpack .../libunwind8_1.6.2-3_arm64.deb ...
#0 4.057 Unpacking libunwind8:arm64 (1.6.2-3) ...
#0 4.077 Setting up libunwind8:arm64 (1.6.2-3) ...
#0 4.079 Processing triggers for libc-bin (2.36-9+deb12u7) ...
#0 5.232 POWERSHELL_VERSION=7.4.3
#0 8.674 SHA256: 6ba8ae0dedabbbfd4bf4a8f48d8819ec504ef092d4c3df120ff6565e542e39b1
#0 8.888 sha256sum: WARNING: 1 computed checksum did NOT match
#0 8.888 powershell-7.4.3-linux-arm64.tar.gz: FAILED
#0 8.888 ERROR: Feature "PowerShell" (ghcr.io/devcontainers/features/powershell) failed to install! Look at the documentation at https://github.com/devcontainers/features/tree/main/src/powershell for help troubleshooting this error.
------
Dockerfile.extended:53
--------------------
  52 |     
  53 | >>> RUN --mount=type=bind,from=dev_containers_feature_content_source,source=powershell_3,target=/tmp/build-features-src/powershell_3 \
  54 | >>>     cp -ar /tmp/build-features-src/powershell_3 /tmp/dev-container-features \
  55 | >>>  && chmod -R 0755 /tmp/dev-container-features/powershell_3 \
  56 | >>>  && cd /tmp/dev-container-features/powershell_3 \
  57 | >>>  && chmod +x ./devcontainer-features-install.sh \
  58 | >>>  && ./devcontainer-features-install.sh \
  59 | >>>  && rm -rf /tmp/dev-container-features/powershell_3
  60 |     
--------------------
error: failed to solve: process "/bin/sh -c cp -ar /tmp/build-features-src/powershell_3 /tmp/dev-container-features  && chmod -R 0755 /tmp/dev-container-features/powershell_3  && cd /tmp/dev-container-features/powershell_3  && chmod +x ./devcontainer-features-install.sh  && ./devcontainer-features-install.sh  && rm -rf /tmp/dev-container-features/powershell_3" did not complete successfully: exit code: 1
[23688 ms] Error: Command failed: docker buildx build --load --build-context dev_containers_feature_content_source=/tmp/devcontainercli-root/container-features/0.62.0-1718955123548 --build-arg _DEV_CONTAINERS_BASE_IMAGE=mcr.microsoft.com/devcontainers/python:3.11 --build-arg _DEV_CONTAINERS_IMAGE_USER=root --build-arg _DEV_CONTAINERS_FEATURE_CONTENT_SOURCE=dev_container_feature_content_temp --target dev_containers_target_stage -f /tmp/devcontainercli-root/container-features/0.62.0-1718955123548/Dockerfile.extended -t vsc-azure-search-openai-demo-2868f436e41d9d185e18eea712c952c50496a000d57ac9f2d1ff61db04503393-features /tmp/devcontainercli-root/empty-folder
[23688 ms]     at ptA (/root/.vscode-remote-containers/dist/dev-containers-cli-0.369.0/dist/spec-node/devContainersSpecCLI.js:464:1260)
[23689 ms]     at NH (/root/.vscode-remote-containers/dist/dev-containers-cli-0.369.0/dist/spec-node/devContainersSpecCLI.js:464:1002)
[23689 ms]     at processTicksAndRejections (node:internal/process/task_queues:96:5)
[23689 ms]     at async KtA (/root/.vscode-remote-containers/dist/dev-containers-cli-0.369.0/dist/spec-node/devContainersSpecCLI.js:481:3692)
[23689 ms]     at async eB (/root/.vscode-remote-containers/dist/dev-containers-cli-0.369.0/dist/spec-node/devContainersSpecCLI.js:481:4807)
[23689 ms]     at async hrA (/root/.vscode-remote-containers/dist/dev-containers-cli-0.369.0/dist/spec-node/devContainersSpecCLI.js:661:13373)
[23689 ms]     at async lrA (/root/.vscode-remote-containers/dist/dev-containers-cli-0.369.0/dist/spec-node/devContainersSpecCLI.js:661:13114)
[23697 ms] Exit code 1
[23698 ms] Start: Run: docker rm -f 65aaa04aa01a521024cb9a61dfc9742a6ee96037abb4506f5a66ac29e0f4652c
[23702 ms] Command failed: node /root/.vscode-remote-containers/dist/dev-containers-cli-0.369.0/dist/spec-node/devContainersSpecCLI.js up --container-session-data-folder /tmp/devcontainers-a561c29f-5151-478f-a4f3-48625abc0d111718955119614 --workspace-folder /workspaces/azure-search-openai-demo --workspace-mount-consistency cached --id-label vsch.local.repository=https://github.com/azure-samples/azure-search-openai-demo --id-label vsch.local.repository.volume=azure-search-openai-demo-89f6766921d19d16d6a45e0edbfd51524ae5339d896354f7663c92e61e6e239a --id-label vsch.local.repository.folder=azure-search-openai-demo --id-label devcontainer.config_file=/workspaces/azure-search-openai-demo/.devcontainer/devcontainer.json --log-level debug --log-format json --config /workspaces/azure-search-openai-demo/.devcontainer/devcontainer.json --override-config /tmp/devcontainer-50671679-b879-4704-9ca0-992e0c30199d.json --default-user-env-probe loginInteractiveShell --mount type=volume,source=azure-search-openai-demo-89f6766921d19d16d6a45e0edbfd51524ae5339d896354f7663c92e61e6e239a,target=/workspaces,external=true --mount type=volume,source=vscode,target=/vscode,external=true --skip-post-create --update-remote-user-uid-default off --mount-workspace-git-root
[23703 ms] Exit code 1
[23758 ms] Container server terminated (code: 137, signal: null).

### Expected/desired behavior
>

### OS and Version?
> Mac pro M2 - Sonoma

### azd version?
> run `azd version` and copy paste here.

### Versions
>

### Mention any other details that might be useful

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by Lawndemon** [comment]
bump this one - I'm getting this out of nowhere starting today. Everything was great yesterday and, with no changes, this cropped up on a clean build this morning. 

I tested the docker setup by spinning up simple containers and it's all good.

**Comment by pamelafox** [comment]
I saw this failure in our Codespace pre-build CI as well. I'll try to replicate.

**Comment by pamelafox** [comment]
See a fix in #1752 - removing the powershell feature fixes it.

**Comment by pope-llamas0g** [comment]
Thank you so much!
